### PR TITLE
deal with html overrides

### PR DESCRIPTION
**Pull request by eyaler**
account for html char overrides to allow embedding stringified js template literals in html. see https://stackoverflow.com/a/10081375/664456
, https://https://html.spec.whatwg.org/multipage/parsing.html#table-charref-overrides

hi - i ran your test suite and it passes, however i am not sure how to add new test for the problem this patch solves. this is the first time i am running node.js and mocha so i am not sure about this, but afaiu we would need to run some browser emulation to test that the html override is resolved correctly. what do you think?

### CODE REVIEW

File: src/simple-yenc.js [hunk 1]
```diff
@@ -36,6 +36,10 @@ const decode = (string) => {
       continue;
     }
 
+    if (byte > 255) {
+      byte = 128 + [8364,,8218,402,8222,8230,8224,8225,710,8240,352,8249,338,,381,,,8216,8217,8220,8221,8226,8211,8212,732,8482,353,8250,339,,382,376].indexOf(byte)
```
Comment (eshaz) [inline src/simple-yenc.js:40]: Which character code set are the numbers representing in the array? I'm a bit concerned that these values may be different depending on the character encoding of the source string.

Comment (eyaler) [reply src/simple-yenc.js:40]: html automatically changes byte values 128-160 the these unicode code points. i am expressing them as integers (and not strings) so they can work with custom html charsets. see here: https://html.spec.whatwg.org/multipage/parsing.html#table-charref-overrides

**Review by eshaz — APPROVED** [review]
This looks good. There are a few things I want to do to this before merging and releasing into master, so I'm going to merge this PR into it's own branch.

Thanks for contributing!

**Comment by eyaler** [comment]
thanks! to alleviate your concerns i can try to explore the option of not using a conversion table, but actually have it created dynamically by the browser for the codes 128-159.

**Comment by eshaz** [comment]
@eyaler That would be interesting if you can figure that out, but the conversion table should be fine. Reading the HTML standard document, it seemed like this will not change for other single byte encodings, for HTML at least.

Also, I released this change in version 0.1.0. I used a `Map` rather than an array to select the index for better performance and also added a test.

**Comment by eyaler** [comment]
@eshaz thanks! i see you shifted by 1 and added to 127. was there an issue with 128 or is this just your preference? also if you are using map you can add 65533 which is the override for nul. i know that nul is not allowed, but if you are interested i can open an issue/pr to optionally allow nul and LF, which i am using in my stringified use case. so Y-ENCode? :) the only thing that needs escaping is CR.

**Comment by eshaz** [comment]
> i see you shifted by 1 and added to 127. was there an issue with 128 or is this just your preference?

I did that so the if statement could check for truthyness rather than `undefined` explicitly when determining when to add the value from the map or use the actual character code.

> also if you are using map you can add 65533 which is the override for nul. i know that nul is not allowed, but if you are interested i can open an issue/pr to optionally allow nul and LF, which i am using in my stringified use case. so Y-ENCode? :) the only thing that needs escaping is CR.

Another PR is always welcome. I would ask that the existing test cases pass though before it can be merged. Technically, not escaping `NULL` and `LF` would break compatibility with any yEnc strings encoded with another tool assuming they follow the standard noted here: http://www.yenc.org/general.htm

**Comment by eyaler** [comment]
@eshaz If interested, I ended up implementing my ideas here: https://github.com/eyaler/ztml

**Pull request merged**